includes/plugins/class-newspack-newsletters.php

```diff
@@ -210,7 +282,7 @@ function( $key ) {
 		}
 
 		// Parse full name into first + last for MC, which stores these as separate merge fields.
-		if ( method_exists( 'Newspack_Newsletters', 'service_provider' ) && 'mailchimp' === \Newspack_Newsletters::service_provider() ) {
+		if ( 'mailchimp' === \get_option( 'newspack_newsletters_service_provider', false ) ) {
```

Comment:
Had to change this to use the raw option in order to get unit tests to play nicely.